src/main/java/dk/kb/image/api/v1/impl/AccessApiServiceImpl.java

```diff
@@ -395,6 +408,77 @@ public StreamingOutput iIPImageRequest(
         }
     }
      
+    /* Is request classified as a thumbnail or fullsize for IIIF requests.
+     * 
+     * This implementation is very conservative and will determine thumbnail also if most non size-parameters are defined.
+     * It is better to be conservative and later loosen up than giving too much control over thumbnail extraction.
+     * 
+     * Will be full size if any other parameters than FIF and CVT is defined. Also WID and HEI must be below a defined limit in the configuration or it will also be fullsize.   
+     */
+    private boolean isThumbnailIIIF(String identifier, String region, String size, String rotation, String quality, String format) {
+         if (!"full".equals(region)  ||  !rotation.equals("0") || !quality.equals("default")) {        	         	
+        	 log.debug("Fullsize for IIIF request since a custom parameter was defined");             
+             return false;                
+         }
+         else if (size == null) {
+             log.debug("Fullsize for IIIF request since size parameter was not defined");
+             return false;
+         }                           
+         
+            //Only allow size as "w,h" parameter. Etc. "100,100". Value is default 'max' when requesting the full image.
+         String[] tokens = size.substring(1).split(","); //First character is !
```

Comment:
If the thumbnail maxima are `75×100` and the input is `975×100`, it will be accepted as a thumbnail.

Reply:
Can not reproduce. I have disabled fullsize now in licensemodule.
http://devel11.statsbiblioteket.dk:10001/ds-image/v1/IIIF/%2FDAMJP2%2FDAM%2FSamlingsbilleder%2F0000%2F750%2F978%2FDNF_1955-00298_00092/full/!150%2C150/0/default.jpg

Changing from 150,150 to 975,100 will not alllow image.

Reply:
So first character not always '!' . Many traps:
Fixed: http://devel11.statsbiblioteket.dk:10001/ds-image/v1/IIIF/%2FDAMJP2%2FDAM%2FSamlingsbilleder%2F0000%2F750%2F978%2FDNF_1955-00298_00092/full/750%2C150/0/default.jpg
(I have disable full-size in licensemodule)

Reply:
The code is now
```Java
String[] tokens= null;
         if (size.startsWith("!")) {
             tokens = size.substring(1).split(","); //First character is !. Example: !150,160
         }
         else {
             tokens = size.split(",");  //  Example: 150,150 
         }
```
and does handle all the ways one can specify the size. `^!w,h` is a very clear and usable way of specifying a thumbnail and is not supported. Please take a look at the list of possible values for  `size` at http://devel11.statsbiblioteket.dk:10001/ds-image/api/#/Access/IIIFImageRequest (or in the IIIF standard).

Secondarily, this method only allows calls where both width & height are specified. That's fair as it is a conservative way of validating. However, in the `isThumbnailIIP`-method it is allowed to only state either width or height. So the behaviour between the two methods is inconsistent.

Reply:
Nice documentation of the regexp! And it's fine now as it acts as a whitelist - we can always extend with more cases if it becomes relevant.